integration-tests/tests/src/tests/objects.ts

```diff
@@ -257,4 +258,66 @@ describe("Realm objects", () => {
       expect(persons.length).equals(1);
     });
   });
+  describe("listeners on linked objects", () => {
```

Comment:
It seems to me like you're adding listeners on collections and not on individual objects?

```suggestion
  describe("collection listeners", () => {
```